### DISCUSSION TITLE
Is there gonna be a openAPI editor in the Editor?

### DISCUSSION THREAD

**Discussion by Buzut** [scalar/scalar#3858] [Ideas]
Currently, the editor only allows to import an openAPI reference.
Insomnia allows to write/edit the spec right from the editor. An experience close to what "OpenAPI (Swagger) Editor" does in VSCode would be mint! All the API related work grouped in one tool.

**Comment by hanspagel** [comment]
OMG, great suggestion. But what are you using? https://docs.scalar.com, `@scalar/api-reference-editor` or the Scalar App?

**Reply by Buzut** [reply]
I'm using both docs.scalar.com (there's the real time editor there for sure [though there isn't any code hints]) and Scalar App. But I was specifically talking about the Scalar App (sorry my initial post might not have been this clear!).

Now being able to edit/update (and sync with either remote Git or scalar.com [but that can be done through Git anyways]) the openAPI definition right from the Scalar App would be a huge improvement in workflow 🙌

Here's a screen of the Insomnia interface (can do wayyy better IMHO). 

![Screenshot 2024-11-12 at 11 43 07](https://github.com/user-attachments/assets/404586e8-8e33-4873-b36d-a46938900159)

**Reply by Buzut** [reply]
What I could also see as a related feature, is the openAPI definition generation from a collection as we have already all the tools to define an exhaustive one:
- environment allows to name the API (title and description)
- we have a collection of requests, with methods and exemples for different queries/responses
- and we have tags to group them 🎉

**Comment by Buzut** [comment]
To add some new info, I found out that Bruno can also do that: https://docs.usebruno.com/get-started/import-export-data/export-collections

Now for Scalar, imagine being able to export this straight from the App and sync it with scalar.com!
